activesupport/lib/active_support/parameter_filter.rb

```diff
@@ -28,12 +28,27 @@ module ActiveSupport
   class ParameterFilter
     FILTERED = "[FILTERED]" # :nodoc:
 
-    def initialize(filters = [])
+    # Create instance with given filters. Supported type of filter is +String+, +Regexp+, and +Proc+.
+    # Other type of filters are treated as +String+ using +to_s+.
+    # For +Proc+ filters, key, value, and optional original hash is passed to block arguments.
+    #
+    # ==== Options
+    #
+    # * <tt>:expose</tt> - Values that will not be masked even if key matches one of +String+ or +Regex+ filters.
+    #   +Proc+ filters are not affected by this value. Defaults to <tt>[]</tt>.
+    def initialize(filters = [], expose: [])
```

Comment:
The `:expose` option is only implemented a string filter, not implemented yet regex and proc filters, how this option should be implemented is out of scope in this PR though.

The reason that this option is needed is to make filter_attributes behave differently than http params filter.

My preference is consistent behavior and less code, so the extra code for the purpose makes less sense to me.

Reply:
ah I've misreading the doc, the `:expose` option behave filtering keys by `filters` first, then excluding values of the matched keys if values exactly matches in the `expose` collection.

Reply:
At least we could implement more less code for the purpose even without this option.

```diff
diff --git a/activerecord/lib/active_record/core.rb b/activerecord/lib/active_record/core.rb
index b83facf4b7..47fc382d70 100644
--- a/activerecord/lib/active_record/core.rb
+++ b/activerecord/lib/active_record/core.rb
@@ -540,7 +540,8 @@ def pretty_print(pp)
               pp.text attr_name
               pp.text ":"
               pp.breakable
-              value = inspection_filter.filter_param(attr_name, read_attribute(attr_name))
+              value = read_attribute(attr_name)
+              value = inspection_filter.filter_param(attr_name, value) unless value.nil?
               if value == ActiveSupport::ParameterFilter::FILTERED
                 pp.text value
               else
@@ -598,7 +599,7 @@ def custom_inspect_method_defined?
       end
 
       def inspection_filter
-        @inspection_filter ||= ActiveSupport::ParameterFilter.new(self.class.filter_attributes, expose: [nil])
+        @inspection_filter ||= ActiveSupport::ParameterFilter.new(self.class.filter_attributes)
       end
   end
 end
```